test/new_tests/test_batch_read.py

```diff
@@ -0,0 +1,129 @@
+import pytest
+
+from aerospike_helpers.batch.records import BatchRecords
+from aerospike_helpers.expressions import base as exp
+from aerospike import exception as e
+
+from .test_base_class import TestBaseClass
+
+
+class TestBatchRead(TestBaseClass):
+    @pytest.fixture(autouse=True)
+    def setup(self, request, connection_with_config_funcs):
+        as_connection = connection_with_config_funcs
+
+        if self.server_version < [6, 0]:
+            pytest.mark.xfail(reason="Servers older than 6.0 do not support batch read.")
+            pytest.xfail()
+
+        self.test_ns = "test"
+        self.test_set = "demo"
+        self.keys = []
+        self.keys_to_expected_bins = {}
+        self.batch_size = 5
+
+        for i in range(self.batch_size):
+            key = ("test", "demo", i)
+            rec = {
+                "count": i,
+                "ilist_bin": [
+                    i,
+                    1,
+                    2,
+                    6,
+                ],
+                "imap_bin": {
+                    1: 1,
+                    2: 2,
+                    3: 6,
+                },
+            }
+            as_connection.put(key, rec)
+            self.keys.append(key)
+            self.keys_to_expected_bins[key] = rec
+
+        def teardown():
+            for i in range(self.batch_size):
+                key = ("test", "demo", i)
+                as_connection.remove(key)
+
+        request.addfinalizer(teardown)
+
+    def test_batch_read_with_policy(self):
+        expr = exp.Eq(exp.IntBin("count"), 1).compile()
+        self.as_connection.batch_read(self.keys, policy={"respond_all_keys": False, "expressions": expr})
```

Comment:
Suggestion, not blocking. A nice way to test that the policy makes it to the C client and server could be to pass an expression that filters out all records, then check that the batch_result is empty (or that they all have non 0 error codes, I forget how that works)

Reply:
I only added this check because I wanted code coverage for when the expressions memory gets freed in the cleanup section of batch read.

Reply:
Improved the test with your suggestion